src/Authorization.AspNetCore/AuthorizationValidationRule.cs

```diff
@@ -102,100 +103,21 @@ private async Task AuthorizeAsync(INode node, IProvideMetadata type, ValidationC
 
                 foreach (var result in authorizationResults)
                 {
-                    AddValidationError(node, context, operationType, result);
+                    if (!result.Succeeded)
+                        AddValidationError(node, context, operationType, result);
                 }
             }
-
-            static void AddValidationError(INode node, ValidationContext context, OperationType? operationType, AuthorizationResult result)
-            {
-                if (!result.Succeeded)
-                {
-                    var error = new StringBuilder("You are not authorized to run this ")
-                                .Append(GetOperationType(operationType))
-                                .Append(".");
-
-                    foreach (var failure in result.Failure.FailedRequirements)
-                    {
-                        AppendFailureLine(error, failure);
-                    }
-
-                    context.ReportError(new ValidationError(context.OriginalQuery, "6.1.1", error.ToString(), node == null ? Array.Empty<INode>() : new INode[] { node })
-                    {
-                        Code = "authorization",
-                    });
-                }
-            }
-        }
-
-        private static string GetOperationType(OperationType? operationType)
-        {
-            return operationType switch
-            {
-                OperationType.Query => "query",
-                OperationType.Mutation => "mutation",
-                OperationType.Subscription => "subscription",
-                _ => "operation",
-            };
         }
 
-        private static void AppendFailureLine(StringBuilder error, IAuthorizationRequirement authorizationRequirement)
+        /// <summary>
+        /// Adds an authorization failure error to the document response
+        /// </summary>
+        protected void AddValidationError(INode node, ValidationContext context, OperationType? operationType, AuthorizationResult result)
```

Comment:
Not virtual? So how to change  `new AuthorizationError` to something other?

Reply:
`protected` infers `virtual`.  It is illegal to write `protected virtual`.

Reply:
(or at least the code style rules disallow it)

Reply:
I read your comment several times. It feels like 2 + 2 = 5 )

Reply:
Yeah, the xml comments were spur-of-the-moment.  I figured you could tweak them up a bit, but I wanted to jot down something so that you could change them via github code review.

Reply:
I talked about this: 
![изображение](https://user-images.githubusercontent.com/21261007/102829667-ac80ba80-43f8-11eb-8b42-d61be26f2927.png)

I almost fell off my chair.

Reply:
oh - lol - i certainly could be wrong!

Reply:
no surprise, I am wrong!  I felt sure I tried that first, and had to change it.  I must be losing my mind lol